infrastructure/eid-wallet/src/lib/fragments/IdentityCard/IdentityCard.svelte

```diff
@@ -0,0 +1,108 @@
+	Upload03Icon,
+	ViewIcon,
+} from "@hugeicons/core-free-icons";
+import { HugeiconsIcon } from "@hugeicons/svelte";
+import type { HTMLAttributes } from "svelte/elements";
+
+interface userData {
+	[fieldName: string]: string;
+}
+interface IIdentityCard extends HTMLAttributes<HTMLElement> {
+	variant?: "eName" | "ePassport" | "eVault";
+	userId?: string;
+	viewBtn?: () => void;
+	shareBtn?: () => void;
+	userData?: userData;
+	totalStorage?: number;
+	usedStorage?: number;
+}
+
+let {
+	variant = "eName",
+	userId,
+	viewBtn,
+	shareBtn,
+	userData,
+	totalStorage = 0,
+	usedStorage = 0,
+	...restProps
+}: IIdentityCard = $props();
+const state = $state({
+	progressWidth: "0%",
+});
+
+$effect(() => {
+	state.progressWidth =
+		usedStorage > 0 ? `${(usedStorage / totalStorage) * 100}%` : "0%";
+});
+</script>
+
+<div {...restProps} class="relative {variant === 'eName' ? "bg-black-900" : variant === 'ePassport' ? "bg-primary" : "bg-gray"}  rounded-xl w-full min-h-[150px] text-white shadow-lg overflow-hidden">
+    <div class="w-full h-full pointer-events-none flex gap-13 justify-end absolute right-15 bottom-20">
+        <div class="w-10 {variant === 'eVault' ? "bg-white/40" : "bg-white/10"} h-[300%] rotate-40"></div>
+        <div class="w-10 {variant === 'eVault' ? "bg-white/40" : "bg-white/10"} h-[300%] rotate-40"></div>
+    </div>
+    <div class="p-5 flex flex-col gap-2">
+        <div class="flex justify-between">
+            {#if variant === 'eName'}  
+                <HugeiconsIcon size={30} strokeWidth={2} color="var(--color-secondary)" icon={CheckmarkBadge02Icon} />
+                <!-- svelte-ignore a11y_click_events_have_key_events -->
+                <div class="flex gap-3 items-center">
+                    <!-- svelte-ignore a11y_click_events_have_key_events -->
+                    <!-- svelte-ignore a11y_no_static_element_interactions -->
+                    <span class="flex justify-start" onclick={shareBtn}>
+                    <HugeiconsIcon size={30} strokeWidth={2} color="white" icon={Upload03Icon} />
+                    </span>
+                    <!-- svelte-ignore a11y_click_events_have_key_events -->
+                    <!-- svelte-ignore a11y_no_static_element_interactions -->
+                    <span class="flex justify-start" onclick={viewBtn}>
+                        <HugeiconsIcon size={30} strokeWidth={2} color="white" icon={ViewIcon} />
+                    </span>
+                </div>
+            {:else if variant === 'ePassport'}
+                <p class="bg-white text-black flex items-center rounded-4xl px-5 py-2 text-xs font-semibold">HIGH SECURITY</p>
+                <!-- svelte-ignore a11y_click_events_have_key_events -->
+                <!-- svelte-ignore a11y_no_static_element_interactions -->
+                <span class="flex justify-start" onclick={viewBtn}>
+                    <HugeiconsIcon size={30} strokeWidth={2} color="white" icon={ViewIcon} />
+                </span>
```

Comment:
we can technically change the span to `button` and just have it display inline instead of display block and it achieves the same thing but with more HTMLiness